source/containers/DynamicCardRenderer/DynamicCardRenderer.tsx

```diff
@@ -82,20 +71,20 @@ const renderCardComponent = (component: CardComponent, navigation: any) => {
 
   // Treat buttons separately, because they have some more complicated behavior
   if (component.type === 'button') {
-    const { icon, iconPosition, text, action } = component;
+    const { icon, iconPosition, text } = component;
     let onClick: () => void = () => null;
-    switch (action.type) {
+    switch (component.action) {
       case 'email':
-        onClick = () => { launchEmail(action.email)};
+        onClick = () => { launchEmail(component.email)};
         break;
       case 'phone':
-        onClick = () => { launchPhone(action.phonenumber)};
+        onClick = () => { launchPhone(component.phonenumber)};
         break;
       case 'navigate':
-        onClick = () => { if (navigation?.navigate) navigation.navigate(action.screen) }; // TODO think about sending parameters here
+        onClick = () => { if (navigation?.navigate) navigation.navigate(component.screen) }; // TODO think about sending parameters here
         break;
       case 'url':
-        onClick = () => { Linking.openURL(action.url)};
+        onClick = () => { Linking.openURL(component.url)};
         break;
     }
```

Comment:
Avoid Switch Statements
https://youtu.be/7EmboKQH8lM?t=4560

TODO = NEVER DO.... :)